fastly/server.go

```diff
@@ -0,0 +1,244 @@
+package fastly
+
+import (
+	"fmt"
+	"sort"
+	"time"
+)
+
+// ServerType represents a server response from the Fastly API.
+type Server struct {
+	ServiceID string `mapstructure:"service_id"`
+	PoolID    string `mapstructure:"pool_id"`
+	ID        string `mapstructure:"id"`
+
+	Address      string     `mapstructure:"address"`
+	Comment      string     `mapstructure:"comment"`
+	Weight       uint       `mapstructure:"weight"`
+	MaxConn      uint       `mapstructure:"max_conn"`
+	Port         uint       `mapstructure:"port"`
+	Disabled     bool       `mapstructure:"disabled"`
+	OverrideHost string     `mapstructure:"override_host"`
+	CreatedAt    *time.Time `mapstructure:"created_at"`
+	DeletedAt    *time.Time `mapstructure:"deleted_at"`
+	UpdatedAt    *time.Time `mapstructure:"updated_at"`
+}
+
+// serversByAddress is a sortable list of servers.
+type serversByAddress []*Server
+
+// Len, Swap, and Less implement the sortable interface.
+func (s serversByAddress) Len() int      { return len(s) }
+func (s serversByAddress) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s serversByAddress) Less(i, j int) bool {
+	return s[i].Address < s[j].Address
+}
+
+// ServerOptionsInput is used to define the optional fields for a server. It is
+// used as part of create and update server operations.
+type ServerOptionsInput struct {
+	Comment      string `form:"comment,omitempty"`
+	Weight       uint   `form:"weight,omitempty"`
+	MaxConn      uint   `form:"max_conn,omitempty"`
+	Port         uint   `form:"port,omitempty"`
+	Disabled     bool   `form:"disabled,omitempty"`
+	OverrideHost string `form:"override_host,omitempty"`
+}
```

Comment:
This doesn't seem to be used anywhere, please delete if so?

Reply:
Good catch! This should have been removed. I was attempting use a single struct to define common fields between the Server create and update operations but I don't think the serializer supports using embedded structs.

Removed via e088bc5